### ISSUE TITLE
Bug: [no-extra-parens] False positive when calling generic function with type parameter with parentheses

### ISSUE THREAD

**Issue by xsjcTony** [typescript-eslint/typescript-eslint#5488]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&jsx=true&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3gMwFzwB4AVAPgAoAPfIgSngF4T4oUBPAWACgvsCyUArgFsARiBjwAPvEwwAligDmNANoBdcioDkULQBp4ARjU0gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDWsX7RETVBkiJo0DtEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A

### Repro Code

```typescript
declare const f: <T>(x: T) => any

f<(number | string)[]>(['a', 1])
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-extra-parens": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

No error

### Actual Result

@typescript-eslint/no-extra-parens: Unnecessary parentheses around expression. 3:23 - 3:24

### Additional Info

Related to #2314

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.33.1` |
| `@typescript-eslint/parser`            | `5.33.1` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.22.0` |
| `node`                                 | `16.16.0` |

**Comment by fede4rodriguez** [comment]
added
